### PR TITLE
ebook converter scripts

### PR DESCRIPTION
**Pull request by entorb**
as discussed here my scripts for conversion the .tex files into html and epub formats

### CODE REVIEW

File: ebook/2_html2epub.cmd [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@echo off
+rem by Torben Menke https://entorb.net
+title HPMOR: html to epub
+
+rem use calibre instead of pandoc, as pandoc looses the css style
```
Comment (rrthomas) [inline ebook/2_html2epub.cmd:5]: "loses" not "looses"!

File: ebook/2_html2epub.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# by Torben Menke https://entorb.net
+# use calibre instead of pandoc, as pandoc looses the css style
```
Comment (rrthomas) [inline ebook/2_html2epub.sh:2]: "loses" not "looses"!

File: ebook/2_html2epub.cmd [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@echo off
```
Comment (rrthomas) [inline ebook/2_html2epub.cmd:1]: Please can we not have a `.cmd` script that does the same thing as a shell script? I think it's not unreasonable these days on Windows to run a shell script; there are plenty of ways to do so.

File: ebook/1_latex2html.py [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+    return out
+
+
+def convert_parsel(s: str) -> str:
+    s = s.replace("ss", "ß").replace("s", "ss").replace("ß", "sss")
+    out = f'<span class="parsel">{s}</span>'
+    return out
+
+
+# def convert_footnotes(s: str, authorsnote: bool = False) -> str:
```
Comment (rrthomas) [inline ebook/1_latex2html.py:622]: Why is this code left commented out here?

**Review by rrthomas — COMMENTED** [review]
Thanks for this, impressive work!

How is the `title-en.jpg` image generated? It doesn't seem right to have it committed; if possible it should be made so that it gets updated if the title page does.

I see that the conversion script refers to various places in the text explicitly; should those places have comments pointing to the script so that they can be more easily kept in sync?

File: ebook/2_html2epub.cmd (hunk 1 above)
Comment (entorb) [reply ebook/2_html2epub.cmd:5]: thanks, fixed.

File: ebook/2_html2epub.cmd (hunk 3 above)
Comment (entorb) [reply ebook/2_html2epub.cmd:1]: ok, you are the boss ;-)

File: ebook/2_html2epub.sh (hunk 2 above)
Comment (entorb) [reply ebook/2_html2epub.sh:2]: thank, fixed

File: ebook/1_latex2html.py (hunk 4 above)
Comment (entorb) [reply ebook/1_latex2html.py:622]: It was a try to use a footnote feature. I discovered too late, that it only works in epub V3.
At [calibre ](https://manual.calibre-ebook.com/generated/en/ebook-convert.html#epub-output-options) it says 
> The version of the EPUB file to generate. EPUB 2 is the most widely compatible, only use EPUB 3 if you know you actually need it.
So I decided not to use it, and to produce a more compatible EPUB 2 file and use inline author comments instead.
I added a comment to the code.

**Comment by entorb** [comment]
thanks for your reply and good suggestions. 

concerning the title-en.jpg: I just took a screenshot of the PDF and slightly modified it using GIMP.
(just removed the url, as it looked a bit strange on a title page.)
Do you have an elegant idea of how to keep in sync with the pdf? I could have written some PDF download and PDF extract logic, but that seemed a bit overkill for this kind of static page.

Concerning the "manual" fixing of the sources: I thought a but doing slight modifications in the .tex part, but thought I would be faster just handling it in Python side. I now improved the warning output in case after the conversion latex->html there are latex comments left.

**Comment by rrthomas** [comment]
Thanks very much for the updates; please could you rebase on master and squash to a single commit again?

Automatic image extraction: I would do this with imagemagick from the built PDF.

I'm not sure I quite understand about how you keep the LaTeX sources and converter code in sync: if one of the bits of LaTeX referred to in the converter is changed, how does the person changing the LaTeX know that the converter may need updating?

**Comment by entorb** [comment]
Sorry, I do not have much experience in collaborating in Git. Is there a way to rebase my branch "ebook" to master and merge all my commits into just one? I created a new branch "ebookV2", but had to discover, that it seems not to be possible to change the source branch of a PR. Shall I create a new PR from my new branch?

**Comment by entorb** [comment]
regarding the title page of the ebook: yes imagemagick would work, but for the ebook cover I prefer a grayscale image (based on page 3 of the PDF), but I do not like to have the URL etc in the bottom part of the page, so I would rather keep the image I have I prepared.

**Comment by rrthomas** [comment]
Those are both design changes that have little to do with the medium; I'd rather not have arbitrary differences like that. Of course, your design ideas may well be improvements! I'm not saying it's necessarily better as it is. But I would still prefer that the ePUB and PDF versions be the same by default. How about making your design changes to the LaTeX and having an option to select it?

**Comment by entorb** [comment]
oki, I not switched to extracting the color titlepage (page 1) of the PDF
(took a little extra work, since imagemagick nowadays complains about extracting pages from pdf, so used ghostscript instead)
result is in a new branch [ebookV3](here: https://github.com/entorb/hpmor-pdflatex-en/tree/ebookV3)

**Comment by entorb** [comment]
just updated to the latest version of the scripts. Additionally, I added the 6 parts as <h1> and shifted the chapters to <h2>

**Pull request closed without merging**